src/pages/Shop/hooks/useGetShopReview.ts

```diff
@@ -0,0 +1,32 @@
+import { useSuspenseQuery } from '@tanstack/react-query';
+import type { UnorderableShopReviewsResponse } from '@/api/shop/entity';
+import type { SortType } from '@/pages/Shop/components/SortModal';
+import { getShopTotalReview } from '@/api/shop';
+import { mapSortTypeToServer } from '@/util/ts/ReviewSorterMapper';
+
+interface UseShopReviewOptions {
+  page?: number;
+  limit?: number;
+  sort?: SortType;
+}
+
+export const useShopReview = (shopId: string, options?: UseShopReviewOptions) => {
+  const page = options?.page ?? 1;
+  const limit = options?.limit ?? 10;
+  const sort = options?.sort ?? 'LATEST';
+
+  const serverSorter = mapSortTypeToServer(sort);
+
+  const query = useSuspenseQuery<UnorderableShopReviewsResponse>({
+    queryKey: ['shopTotalReview', shopId, page, limit, serverSorter],
+    queryFn: () =>
+      getShopTotalReview({
+        shopId,
+        page,
+        limit,
+        sorter: serverSorter,
+      }),
+  });
+
+  return query;
+};
```

Comment:
```suggestion
interface UseShopReviewProps {
  shopId: string;
  page?: number;
  limit?: number;
  sort?: SortType;
}

export const useShopReview = ({ shopId, page = 1, limit = 50, sort = 'LATEST' }: UseShopReviewProps) => {
  const query = useSuspenseQuery<UnorderableShopReviewsResponse>({
    queryKey: ['shopTotalReview', shopId, page, limit, sort],
    queryFn: () =>
      getShopTotalReview({
        shopId,
        page,
        limit,
        sorter: sort,
      }),
  });

  return query;
};
```

이렇게 수정하고 사용하는 곳에서 
const { data: totalData } = useShopReview({ shopId, sort });

이렇게 사용하는 건 어떻게 생각하시나요?

Reply:
9a72a2b42e38c9e8369113203447d96514cb6d22
수정했습니다.